uaclient/entitlements/repo.py

```diff
@@ -144,6 +151,43 @@ def operational_status(self):
             return status.ACTIVE, '%s is active' % self.title
         return status.INACTIVE, '%s is not configured' % self.title
 
+    def process_contract_deltas(
```

Comment:
I have a number of comments in here relating to behavioural differences from `enable`. To avoid always having to manually reconcile changes to the two, I wonder if we should do some refactoring so that `enable` and `process_contract_deltas` use the same code?

(A more radical idea would be to refactor things entirely, so that "enable" is really just processing a delta from {} to our initial entitlement.)

What do you think?

Reply:
(A more radical idea would be to refactor things entirely, so that "enable" is really just processing a delta from {} to our initial entitlement.) 

I was leaning toward this suggestion and couldn't decide whether or not to do that in this branch.
I left the comment in the code to imply that we should be doing just that. # TODO: handle initial install deltas where orig_access is empty

I was uncertain whether or not to bite off that chunk of work in this branch, but given your inline comments, this branch will end up ultimately duplicating a lot of that common logic anyway to ensure the delta process behaves the same as green-field enable(). I'll refactor to use the same logic so we don't have to deconcile across different logical paths.